### PR TITLE
US680204: Revisiting annotations

### PR DESCRIPTION
**Pull request by nikhilahuja**
## Description
US680204: Revisiting annotations
- `@bundle` doesn't contain any attribute like name, tags, description
- `@bundle-entity` is changed to `@bundle-hints`
- `@bundle-hints` contains the bundle related attributes like id, guid, name, tags, description

## Checklist
- [x] Pull Request Created
- [x] Unit tests have been added
- [x] Integration/Functional test cases have been written and automated

### CODE REVIEW

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotableEntity.java [hunk 1]
```diff
@@ -50,14 +50,14 @@
         if (annotations != null) {
             AnnotatedEntity<GatewayEntity> annotatedEntity = new AnnotatedEntity(this);
             annotations.forEach(annotation -> {
-                if (ANNOTATION_TYPE_BUNDLE.equalsIgnoreCase(annotation.getType())) {
-                    annotatedEntity.setTags(annotation.getTags());
+                if (BUNDLE.equalsIgnoreCase(annotation.getType())) {
                     annotatedEntity.setEntityType(EntityTypes.ENCAPSULATED_ASSERTION_TYPE);
```
Comment (graju256) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotableEntity.java:54]: Though it is not part of this PR, why did we hard-code this? Rather, it should be something like: this.getType().

File: gateway-developer-plugin/src/test/resources/example-project-generating-environment/src/main/gateway/config/encass.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 encass_env:
   annotations:
     - type: "@bundle"
```
Comment (graju256) [inline gateway-developer-plugin/src/test/resources/example-project-generating-environment/src/main/gateway/config/encass.yml:3]: make this annotation simpler.
type: "@bundle" --> - "@bundle"

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotableEntity.java (hunk 1 above)
Comment (nikhilahuja) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotableEntity.java:54]: This will be taken care in separate story for enabling service bundle

File: gateway-developer-plugin/src/test/resources/example-project-generating-environment/src/main/gateway/config/encass.yml (hunk 2 above)
Comment (nikhilahuja) [reply gateway-developer-plugin/src/test/resources/example-project-generating-environment/src/main/gateway/config/encass.yml]: @graju256 done!

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/entity/AnnotationType.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.HashSet;
+import java.util.Set;
+
+public class AnnotationType {
+
+    /**
+     * Constants for Gateway entity annotation types supported by the plugin.
+     */
+    public static final String BUNDLE = "@bundle";
+    public static final String REUSABLE = "@reusable";
```
Comment (uppoju) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/entity/AnnotationType.java:12]: are we not changing this to shared ?

Comment (nikhilahuja) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/util/entity/AnnotationType.java:12]: Will be taken care of later.
